pkg/gardenlet/controller/shoot/shoot/reconciler_migrate.go

```diff
@@ -239,7 +249,7 @@ func (r *Reconciler) runMigrateShootFlow(ctx context.Context, o *operation.Opera
 		deleteExtensionResources = g.Add(flow.Task{
 			Name:         "Deleting extension resources from the Shoot namespace",
 			Fn:           botanist.DestroyExtensionResourcesInParallel,
-			Dependencies: flow.NewTaskIDs(waitUntilExtensionResourcesMigrated),
+			Dependencies: flow.NewTaskIDs(persistShootState),
```

Comment:
I believe, `deleteExtensionResources` should only be done after `waitUntilExtensionResourcesMigrated`.